scopeguard breaking no_std build

I'm trying to use safer_ffi to handle ffi stuff on an embedded device. i'm building my no_std project for --target thumbv7em-none-eabi, and i've included safer_ffi in cargo.toml with default-features = false, yet when i try to build, i get errors about missing STD from scopeguard. i checked and it seems scopeguard should work without std as well if default features are disabled. Is it possible this feature isn't getting forwarded to the dependencies?

`Compiling scopeguard v1.1.0 error[E0463]: can't find crate for std`

`   Compiling scopeguard v1.1.0
error[E0463]: can't find crate for `std`
  |
  = note: the `thumbv7em-none-eabi` target may not support the standard library
  = note: `std` is required by `scopeguard` because it does not declare `#![no_std]`
  = help: consider building the standard library from source with `cargo build -Zbuild-std`
`

--- a/.github/workflows/gh-pages.yml
+++ b/.github/workflows/gh-pages.yml
@@ -6,6 +6,10 @@ on:
       - master
   pull_request:
 
+concurrency:
+  group: ${{ github.workflow }}-${{ github.ref || github.run_id }}
+  cancel-in-progress: true
+
 env:
   URL_ROOT_PATH: /safer_ffi/
 

--- a/.github/workflows/gh-pages.yml
+++ b/.github/workflows/gh-pages.yml
@@ -14,6 +14,38 @@ env:
   URL_ROOT_PATH: /safer_ffi/
 
 jobs:
+  # Check no_std #
+  check-nostd:
+    name: 'Check `#![no_std]`'
+    runs-on: ubuntu-latest
+    steps:
+      - name: Install Rust toolchain
+        uses: actions-rs/toolchain@v1
+        with:
+          profile: minimal
+          toolchain: nightly-2023-06-30
+          override: true
+          components: rust-src
+
+      - name: Clone repo
+        uses: actions/checkout@v2
+
+      - uses: actions-rs/cargo@v1
+        name: '`cargo +nightly check` without `std` nor `alloc`'
+        with:
+          command: check
+          args: |
+            --no-default-features
+            --target x86_64-unknown-linux-gnu -Zbuild-std=core
+
+      - uses: actions-rs/cargo@v1
+        name: '`cargo +nightly check` without `std` '
+        with:
+          command: check
+          args: |
+            --no-default-features --features alloc
+            --target x86_64-unknown-linux-gnu -Zbuild-std=core,alloc
+
   # == CHECK == #
   check:
     name: Check nightly stable and MSRV
@@ -160,6 +192,7 @@ jobs:
   all-good:
     name: All the required checks
     needs:
+      - check-nostd
       - check
       - check-features
       - build-and-test

--- a/Cargo.toml
+++ b/Cargo.toml
@@ -40,6 +40,7 @@ nightly = []
 alloc = []
 std = [
     "alloc",
+    "scopeguard/use_std",
 ]
 
 proc_macros = [] # Deprecated
@@ -136,6 +137,7 @@ once_cell.version = "1.17.1"
 paste.version = "1.0.12"
 
 scopeguard.version = "1.1.0"
+scopeguard.default-features = false
 
 tokio.optional = true
 tokio.version = "1.26.0"

--- a/Cargo.toml
+++ b/Cargo.toml
@@ -3,7 +3,7 @@ path = "src/_lib.rs"
 
 [package]
 name = "safer-ffi"
-version = "0.1.0"  # Keep in sync
+version = "0.1.1"  # Keep in sync
 authors = [
     "Daniel Henry-Mantilla <daniel.henry.mantilla@gmail.com>",
 ]
@@ -161,7 +161,7 @@ version = "0.0.3"
 
 [dependencies.safer_ffi-proc_macros]
 path = "src/proc_macro"
-version = "=0.1.0"  # Keep in sync
+version = "=0.1.1"  # Keep in sync
 
 [target.'cfg(not(target = "wasm32-unknown-unknown"))'.dependencies]
 libc.version = "0.2.66"

--- a/src/proc_macro/Cargo.toml
+++ b/src/proc_macro/Cargo.toml
@@ -4,7 +4,7 @@ proc-macro = true
 
 [package]
 name = "safer_ffi-proc_macros"
-version = "0.1.0"  # Keep in sync
+version = "0.1.1"  # Keep in sync
 authors = ["Daniel Henry-Mantilla <daniel.henry.mantilla@gmail.com>"]
 edition = "2021"
 

